.github/workflows/pipelines.yml

```diff
@@ -0,0 +1,73 @@
+name: CI
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  lint:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8, 3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox
+    - name: Lint with flake8
+      run: |
+        tox -vv -eflake8
+    - name: Lint with pep8
+      run: |
+        tox -vv -epep8
+
+  container:
+
+    needs: lint
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        fedora-release: [32, 33]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Lint with pylint fedora:${{ matrix.fedora-release }}
+      run: |
+        docker pull fedora:${{ matrix.fedora-release }}
```

Comment:
Have you tried the service `container` option to avoid dealing with `docker` manually? It could cut down the code maintenance. My $0.02 😄 

Reference: https://github.com/dogtagpki/pki/blob/master/.github/workflows/ipa-tests.yml#L16 
Documentation: https://docs.github.com/en/free-pro-team@latest/actions/guides/about-service-containers

Reply:
I looked into that and it seemed more for running a service than executing the tests. Since healthcheck isn't a service I didn't think it really applied.